plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -167,17 +169,20 @@ func defaultPredicates() sets.String {
 		factory.RegisterFitPredicate("GeneralPredicates", predicates.GeneralPredicates),
 
 		// Fit is determined based on whether a pod can tolerate all of the node's taints
-		factory.RegisterFitPredicate("PodToleratesNodeTaints", predicates.PodToleratesNodeTaints),
+		func() string {
```

Comment:
Why is this a function? We already check the feature gate at line 196. Why don't we move the body of this function there and add it to the "if" and a new "else" statement there?